src/main/java/com/conveyal/gtfs/graphql/GTFSGraphQL.java

```diff
@@ -1,41 +1,93 @@
 package com.conveyal.gtfs.graphql;
 
+import com.conveyal.gtfs.graphql.fetchers.JDBCFetcher;
+import graphql.ExecutionInput;
 import graphql.GraphQL;
+import graphql.execution.instrumentation.dataloader.DataLoaderDispatcherInstrumentation;
+import graphql.introspection.IntrospectionQuery;
+import graphql.schema.DataFetchingEnvironment;
+import org.dataloader.DataLoader;
+import org.dataloader.DataLoaderRegistry;
 
 import javax.sql.DataSource;
-import java.sql.Connection;
-import java.sql.SQLException;
-
-/**
- * This provides a GraphQL API around the gtfs-lib JDBC storage.
- * This just makes a Java API with the right schema available, which uses String requests and responses.
- * To make this into a web API you need to wrap it in an HTTP framework / server.
- */
+import java.util.List;
+import java.util.Map;
+
+import static com.conveyal.gtfs.graphql.fetchers.JDBCFetcher.newJdbcDataLoader;
+
 public class GTFSGraphQL {
 
-    private static DataSource dataSource;
+    /**
+     * Execute a graphQL request.  This method creates a new graphQL runner that is able to cache and batch sql queries
```

Comment:
Can you clarify in the comment which particular part of the following code makes it capable of batching the requests?